glusterd2/snapshot/lvm/lvm.go

```diff
@@ -0,0 +1,177 @@
+package lvm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/pborman/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	//CreateCommand is path to lvm create
+	CreateCommand = "/sbin/lvcreate"
+	//RemoveCommand is path to lvm create
+	RemoveCommand = "/sbin/lvremove"
+)
+
+// CommonPrevalidation checks the lvm related validation for snapshot
+func CommonPrevalidation(lvmCommand string) error {
+	fileInfo, err := os.Stat(lvmCommand)
+	if err != nil {
+		return err
+	}
+	switch mode := fileInfo.Mode(); {
+
+	case mode.IsRegular() == false:
+		return err
+	case mode&0111 == 0:
+		return err
+	}
+	return nil
+}
+
+//IsbrickComptable check for lvm comptability for a path
+func IsbrickComptable(brick *brick.Brickinfo) bool {
```

Comment:
Please make this as generic function(`isThinLV(fsname)`) without associating with Brickinfo.